.github/workflows/component.yml

```diff
@@ -0,0 +1,74 @@
+---
+name: Component
+
+on:
+  workflow_call:
+    inputs:
+      component:
+        required: true
+        type: string
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            component:
+              # Flake
+              - '**/*.nix'
+              - 'flake.lock'
+              # Cargo common
+              - 'Cargo.toml'
+              - 'Cargo.lock'
+              # Rust common
+              - 'common/Cargo.toml'
+              - 'common/**/*.rs'
+              - 'streaming-types/Cargo.toml'
+              - 'streaming-types/**/*.rs'
+              # Component
+              - "${{ inputs.component }}/flake.nix"
+              - "${{ inputs.component }}/Cargo.toml"
+              - "${{ inputs.component }}/**/*.rs"
```

Comment:
Should there also be filters for the default.nix files and README.md in each component?

Reply:
Ah, that was supposed to be `default.nix` not `flake.nix` there.

The readme is not required as this has no influence over the binary.